keel-web/src/test/kotlin/com/netflix/spinnaker/keel/apidocs/ApiDocCompatibilityTests.kt

```diff
@@ -60,11 +75,24 @@ class ApiDocCompatibilityTests {
       dynamicTest("example delivery config ${it.substringAfterLast("/")} is valid") {
         val messages = schema.validate(loadExample(it))
         expectThat(messages)
-          .describedAs("example delivery config")
+          .describedAs("validation messages for ${it.substringAfterLast("/")}")
           .isValid()
       }
     }
 
+  @TestFactory
+  fun invalidConfigsAreInvalid(): List<DynamicTest> =
+    listOf(
+      "/examples/wrong-resource-kind.yml"
+    ).map {
+      dynamicTest("example delivery config ${it.substringAfterLast("/")} is invalid") {
+        val messages = schema.validate(loadExample(it))
+        expectThat(messages)
+          .describedAs("validation messages for ${it.substringAfterLast("/")}")
+          .isNotValid()
+      }
+    }
+
```

Comment:
A test that checks that a resource where the `kind` and `spec` don't match is rejected. This failed before these changes.